api4/handlers_test.go

```diff
@@ -0,0 +1,86 @@
+	io.WriteString(w, body)
+}
+
+func TestApiHandlerWithGzip(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	WebserverMode := *th.App.Config().ServiceSettings.WebserverMode
+
+	defer func() {
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = WebserverMode })
+	}()
+
+	// WebserverMode = "gzip"
+	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = "gzip" })
+
+	api := Init(th.Server, th.Server.AppOptions, th.Server.Router)
+
+	handler := api.ApiHandler(handlerForGzip)
+
+	// No specified "Accept-Encoding" in req
+	resp1 := httptest.NewRecorder()
+	req1 := httptest.NewRequest("GET", "/api/v4/test", nil)
+
+	handler.ServeHTTP(resp1, req1)
+	header := resp1.Header().Get("Content-Encoding")
+
+	if header != "" {
```

Comment:
You could also use `testify/assert` to shorten this slightly if you wanted
```go
assert.Equals(t, "", resp1.Header().Get("Content-Encoding"))
```

Reply:
Also, copied the styling directly from the `web/handlers_test.go` file. Noted.